lib/src/data/irma_repository.dart

```diff
@@ -505,41 +495,29 @@ class IrmaRepository {
 
   static const _iiabchannel = MethodChannel('irma.app/iiab');
 
-  Future<String> getInAppCredential() {
-    return _inAppCredentialSubject.first.then((state) => state.credentialType);
-  }
-
-  Future<void> processInactivation() async {
-    final curState = await _inAppCredentialSubject.first;
-    if (curState.pendingInactivations > 0) {
-      // If there are still inactivations to be ignored, we ignore
-      // and just decrement count
-      _inAppCredentialSubject.add(curState.copyWith(
-        pendingInactivations: curState.pendingInactivations - 1,
-      ));
-    } else {
-      // Forget about previous opening of browser
-      _inAppCredentialSubject.add(_InAppCredentialState());
-    }
-  }
-
-  // Remember that an inactivation of the app is coming due to opening the browser
-  // for issuance of the given credential type. Opening the browser inactivates the app
-  // (i.e. sets the AppLifecycleState to paused), so we must not react to that the way
-  // we normally do (e.g. go back to the previous app or show the big topleft arrow).
-  void expectInactivationForCredentialType(String type) {
-    _inAppCredentialSubject.add(_InAppCredentialState(pendingInactivations: 1, credentialType: type));
+  Future<List<String>> getCredentialsLaunchedFromStore() {
+    return _credentialStoreSubject.first.then(
+      (state) => state.launchedCredentials,
+    );
   }
 
   Future<void> openIssueURL(BuildContext context, String type) async {
     final lang = FlutterI18n.currentLocale(context)!.languageCode;
-    final translated = await _irmaConfigurationSubject.first
-        .then((irmaConfiguration) => irmaConfiguration.credentialTypes[type]?.issueUrl);
-    final url = translated?.translate(lang, fallback: '') ?? '';
+
+    final irmaConfig = await _irmaConfigurationSubject.first;
+    final cred = irmaConfig.credentialTypes[type]!;
+    final url = cred.issueUrl.translate(lang, fallback: '');
     if (url.isEmpty) {
       throw UnsupportedError('Credential type $type does not have a suitable issue url for $lang');
     }
-    expectInactivationForCredentialType(type);
+
+    if (cred.isInCredentialStore) {
+      final state = await _credentialStoreSubject.first;
+      final updatedLaunchedCredentials = [...state.launchedCredentials, type];
+      _credentialStoreSubject.add(_CredentialStoreState(
+        launchedCredentials: updatedLaunchedCredentials,
+      ));
+    }
```

Comment:
This does not take into account that credentials on the data overview tab do not necessarily have to be in the store.

Reply:
So when `openIssueURL` is called on the data overview tab (for example because we want to refresh a credential that is due to expire), then that credential is not added to `previouslyLaunchedCredentials`? Or do I miss something?

Reply:
The subject state does get updated in `openIssueURL`. Line 536 of `irma_repository.dart `.

Reply:
Ah right. The name `inAppCredentials` did not make that much sense to me. Now I understand what it's doing. Maybe something like `alreadyObtainedCredentials` would be more clear in my view.